ir/instr.cpp

```diff
@@ -1103,75 +1173,88 @@ void FpUnaryOp::rauw(const Value &what, Value &with) {
-  case NearbyInt:    str = "nearbyint "; break;
-  case Round:        str = "round "; break;
-  case RoundEven:    str = "roundeven "; break;
-  case Trunc:        str = "trunc "; break;
-  case Sqrt:         str = "sqrt "; break;
-  }
-
-  os << getName() << " = " << str << fmath << *val;
+  case FAbs:         return "fabs"; break;
+  case FNeg:         return "fneg"; break;
+  case Canonicalize: return "canonicalize"; break;
+  case Ceil:         return "ceil"; break;
+  case Floor:        return "floor"; break;
+  case RInt:         return "rint"; break;
+  case NearbyInt:    return "nearbyint"; break;
+  case Round:        return "round"; break;
+  case RoundEven:    return "roundeven"; break;
+  case Trunc:        return "trunc"; break;
+  case Sqrt:         return "sqrt"; break;
+  }
+  UNREACHABLE();
+}
+
+void FpUnaryOp::print(ostream &os) const {
+  os << getName() << " = " << getOpName() << " " << fmath << *val;
   if (!rm.isDefault())
     os << ", rounding=" << rm;
   if (!ex.ignore())
     os << ", exceptions=" << ex;
 }
 
 StateValue FpUnaryOp::toSMT(State &s) const {
-  expr (*fn)(const expr&, const expr&) = nullptr;
+  function<expr(const expr&, const expr&)> fn = nullptr;
+  function<StateValue(const StateValue&, const Type&)> scalar = nullptr;
   bool bitwise = false;
 
-  switch (op) {
-  case FAbs:
-    bitwise = true;
-    fn = [](const expr &v, const expr &rm) { return v.fabs(); };
-    break;
-  case FNeg:
-    bitwise = true;
-    fn = [](const expr &v, const expr &rm) { return v.fneg(); };
-    break;
-  case Canonicalize:
-    fn = [](const expr &v, const expr &rm) { return v; };
-    break;
-  case Ceil:
-    fn = [](const expr &v, const expr &rm) { return v.ceil(); };
-    break;
-  case Floor:
-    fn = [](const expr &v, const expr &rm) { return v.floor(); };
-    break;
-  case RInt:
-  case NearbyInt:
-    // TODO: they differ in exception behavior
-    fn = [](const expr &v, const expr &rm) { return v.round(rm); };
-    break;
-  case Round:
-    fn = [](const expr &v, const expr &rm) { return v.round(expr::rna()); };
-    break;
-  case RoundEven:
-    fn = [](const expr &v, const expr &rm) { return v.round(expr::rne()); };
-    break;
-  case Trunc:
-    fn = [](const expr &v, const expr &rm) { return v.round(expr::rtz()); };
-    break;
-  case Sqrt:
-    fn = [](const expr &v, const expr &rm) { return v.sqrt(rm); };
-    break;
-  }
+  if (config::is_uf_float()) {
+    scalar = [&](const StateValue &v, const Type &ty) -> StateValue {
+      s.doesApproximation("uf_float", true);
 
-  auto scalar = [&](const StateValue &v, const Type &ty) {
-    return fm_poison(s, v.value, v.non_poison,
-                     [fn](auto &v, auto &rm) {return fn(v, rm);}, ty, fmath, rm,
-                     bitwise, false);
-  };
+      ostringstream name;
+      name << getOpName() << "." << ty;
+      return uf_float(std::move(name).str(), {v}, v.value, fmath, false);
+    };
+  } else {
```

Comment:
could you please reduce the diff size by not doing these huge re-indentation. otherwise it's very hard to review what's going on.
You can have an in afterwards to override the function.

Reply:
What would be your preferred solution for reviewing? Is just not having the existing code indented for now (and then using a second PR that indents the relevant parts) fine? i.e.

```cpp
if (is_uf_float()) {
  <create UF>
} else {
<existing code>
}
```

Reply:
```
existing code
if (is_uf) {
  fn = xpto
}
```

Reply:
I have moved the existing code, the diff is quite a bit more manageable now.